Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -642,6 +642,32 @@ public struct ObjectId: BSONValue, Equatable, CustomStringConvertible, Codable {
 
 }
 
+/// Extension to allow a UUID to be initialized from a Binary BSONValue.
+extension UUID {
+    // TODO: fill the rest of this out for full BSONValue conformance (SWIFT-295).
+
+    internal init(fromBinary binary: Binary) throws {
```

Comment:
a general naming thing we should think about when we write these `from` inits is whether to use `from` or `fromX`. 

Swift standard library tends toward just `from`/`to`, i.e. `init(from decoder: Decoder)`, `encode(to encoder: Encoder)`, etc. 

I think I lean toward just using `from` unless if it is adding more information beyond just the type of the input. for ex. consider the following `Document` initializers:
- `Document.init(fromJSON: Data)`
- `Document.init(fromBSON: Data)`
- `Document.init(fromJSON: String)`

In all cases we are adding more info about the format - it's not just `Data` but BSON data, it's not just a `String` but a JSON string, etc. here, it feels kind of redundant to say `fromBinary`. 

there is the argument that including the type name might make things more readable at the call site, like if I say `UUID(fromBinary: b)`, but I feel like that is more a problem with the user's code if they can't otherwise tell what type `b` is. 

thoughts?

I realize we're not that consistent about this throughout the codebase so I should probably open a ticket about establishing convention here and correcting violations....

Reply:
I think you're right. In statically typed languages like Swift it is pretty redundant to have type information encoded in your labels/variable names, since the compiler will send a much louder and clearer message if there's a mismatch anyways. It also makes refactoring a little harder if you have to change the variable names when you decide to use different types.

changed -> `init(from binary: Binary)`